tests/jerry/es2015/regression-test-issue-2657.js

```diff
@@ -0,0 +1,20 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+try {
+  eval ("var Mixin1 = (superclass) => class extends super.lass {};");
```

Comment:
Good catch! Thanks to this code snippet I've realized this could be done via remove the `PARSER_IS_ARROW_FUNCTION` while parsing the super keyword, since the arrow function inherits the `PARSER_CLASS_HAS_SUPER` flag from the previous context.

Reply:
Please add this test as well.